fda/depth_measures.py

```diff
@@ -0,0 +1,388 @@
+"""Depth Measures Module.
+
+This module includes different methods to order functional data,
+from the center (larger values) outwards(smaller ones)."""
+
+import numpy as np
+from scipy.stats import rankdata
+from .grid import FDataGrid
+
+__author__ = "Amanda Hernando Bernabé"
+__email__ = "amanda.hernando@estudiante.uam.es"
+
+
+def _rank_samples(fdatagrid):
+    """Ranks the he samples in the FDataGrid at each point of discretisation.
+
+    Args:
+        fdatagrid (FDataGrid): Object whose samples are ranked.
+
+    Returns:
+        numpy.darray: Array containing the ranks of the sample points.
+
+    Examples:
+        Univariate setting:
+
+        >>> data_matrix = [[1, 1, 2, 3, 2.5, 2], [0.5, 0.5, 1, 2, 1.5, 1],
+        ...                [-1, -1, -0.5, 1, 1, 0.5], [-0.5, -0.5, -0.5, -1, -1, -1]]
+        >>> sample_points = [0, 2, 4, 6, 8, 10]
+        >>> fd = FDataGrid(data_matrix, sample_points)
+        >>> _rank_samples(fd)
+        array([[[4.],
+                [4.],
+                [4.],
+                [4.],
+                [4.],
+                [4.]],
+        <BLANKLINE>
+               [[3.],
+                [3.],
+                [3.],
+                [3.],
+                [3.],
+                [3.]],
+        <BLANKLINE>
+               [[1.],
+                [1.],
+                [2.],
+                [2.],
+                [2.],
+                [2.]],
+        <BLANKLINE>
+               [[2.],
+                [2.],
+                [2.],
+                [1.],
+                [1.],
+                [1.]]])
+
+        Multivariate Setting:
+
+        >>> data_matrix = [[[[1, 3], [2, 6]], [[23, 54], [43, 76]], [[2, 45], [12, 65]]],
+        ...                [[[21, 34], [8, 16]], [[67, 43], [32, 21]], [[10, 24], [3, 12]]]]
+        >>> sample_points = [[2, 4, 6], [3, 6]]
+        >>> fd = FDataGrid(data_matrix, sample_points)
+        >>> _rank_samples(fd)
+        array([[[[1., 1.],
+                 [1., 1.]],
+        <BLANKLINE>
+                [[1., 2.],
+                 [2., 2.]],
+        <BLANKLINE>
+                [[1., 2.],
+                 [2., 2.]]],
+        <BLANKLINE>
+        <BLANKLINE>
+               [[[2., 2.],
+                 [2., 2.]],
+        <BLANKLINE>
+                [[2., 1.],
+                 [1., 1.]],
+        <BLANKLINE>
+                [[2., 1.],
+                 [1., 1.]]]])
+    """
+    ranks = np.zeros(fdatagrid.shape)
+
+    if fdatagrid.ndim_domain == 1:
```

Comment:
Have you tried to unify both branches with `itertools`. That also would make it work for arbitrary dimensions.

Reply:
implemented